File: app/src/main/java/br/com/levimendesestudos/spidermagazine/activities/MainActivity.java
package br.com.levimendesestudos.spidermagazine.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.GridView;
import android.widget.TextView;
import java.util.List;
import br.com.levimendesestudos.spidermagazine.R;
import br.com.levimendesestudos.spidermagazine.adapters.RevistasListaAdapter;
import br.com.levimendesestudos.spidermagazine.model.Revista;
import br.com.levimendesestudos.spidermagazine.mvp.contracts.MainMVP;
import br.com.levimendesestudos.spidermagazine.mvp.presenter.MainPresenter;

public class MainActivity extends BaseActivity implements MainMVP.View {

    private MainPresenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mPresenter = new MainPresenter(this);
        mPresenter.init();
    }

    @Override
    public void configToolbar() {
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle("");
    }

    @Override
    public void showSemInternet() {
        findViewById(R.id.tvSemInternet).setVisibility(View.VISIBLE);
    }

    @Override
    public void hideSemInternet() {
        findViewById(R.id.tvSemInternet).setVisibility(View.GONE);
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
    }

    @Override
    public void copyRight(String s) {
        ((TextView)findViewById(R.id.tvCopyright)).setText(s);
    }

    @Override
    public void carregarLista(List<Revista> revistas) {
        RevistasListaAdapter adapter = new RevistasListaAdapter(revistas);
        ((GridView)findViewById(R.id.gvRevistas)).setAdapter(adapter);
    }

    @Override
    public void showPbProcessamento() {
        findViewById(R.id.pbProcessamento).setVisibility(View.VISIBLE);
    }

    @Override
    public void hidePbProcessamento() {
        findViewById(R.id.pbProcessamento).setVisibility(View.GONE);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        mPresenter.navigate(item.getItemId());

        return true;
    }

    @Override
    public void callSobreActivity() {
        Intent intent = new Intent(MainActivity.this, SobreActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/br/com/levimendesestudos/spidermagazine/activities/MainActivity.kt
package br.com.levimendesestudos.spidermagazine.activities

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import br.com.levimendesestudos.spidermagazine.R
import br.com.levimendesestudos.spidermagazine.adapters.RevistasListaAdapter
import br.com.levimendesestudos.spidermagazine.koin.ApiSpiderModule
import br.com.levimendesestudos.spidermagazine.model.Revista
import br.com.levimendesestudos.spidermagazine.mvp.contracts.MainMVP
import kotlinx.android.synthetic.main.activity_main.*
import org.koin.android.ext.android.inject
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers
import java.util.HashMap

class MainActivity : BaseActivity(), MainMVP.View {

    val mApi: ApiSpiderModule by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mApi.providesSpiderApi().comics(params())
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe({ hero ->

                if (active()) {
                    carregarLista(hero.revistas)
                    copyRight(hero.copyright)
                    hidePbProcessamento()
                }

            }, {error ->
                Log.e("onError", error.message, error)
                showToast(error.message!!)
            })
    }

    private fun params(): Map<String, String> {
        val params = HashMap<String, String>()

        params["ts"] = "1"
        params["apikey"] = "bb4470a46d0659a43c566ac6056ed48d"
        params["hash"] = "479474cf0a28eac9998960da4d96f06b"

        return params
    }

    override fun configToolbar() {
        setSupportActionBar(toolbar)
        supportActionBar?.title = ""
    }

    override fun showSemInternet() {
        tvSemInternet.visibility = View.VISIBLE
    }

    override fun hideSemInternet() {
        tvSemInternet.visibility = View.GONE
    }

    override fun copyRight(s: String) {
        tvCopyright.text = s
    }

    override fun carregarLista(revistas: List<Revista>) {
        val adapter = RevistasListaAdapter(this, revistas)
        gvRevistas.adapter = adapter
    }

    override fun showPbProcessamento() {
        pbProcessamento.visibility = View.VISIBLE
    }

    override fun hidePbProcessamento() {
        pbProcessamento.visibility = View.GONE
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_profile -> callSobreActivity()
        }

        return true
    }

    override fun callSobreActivity() {
        val intent = Intent(this, SobreActivity::class.java)
        startActivity(intent)
    }
}